File: scripts/docker-updater.js
// EOSJS version 2 used
const { Api, JsonRpc, RpcError } = require('eosjs');
const fetch = require('node-fetch'); 
const { TextEncoder, TextDecoder } = require('util');  
const { JsSignatureProvider } = require('eosjs/dist/eosjs-jssig'); 
const axios = require('axios');
const dotenv = require('dotenv');

dotenv.load();

var chain = process.env.CHAIN;
var priceUrl = "";
var usdpair = "";
var btcpair = "";

// Switch priceURL depending on EOSIO Chain provided via .env
switch (chain) {
  case "wax":
	priceUrl = "https://min-api.cryptocompare.com/data/price?fsym=WAXP&tsyms=BTC,USD";
	usdpair = "waxpusd";
	btcpair = "waxpbtc";
    break;
  case "eos":
	priceUrl = "https://min-api.cryptocompare.com/data/price?fsym=EOS&tsyms=BTC,USD";
	usdpair = "eosusd";
	btcpair = "eosbtc";
    break;
  default:
	var priceUrl = "https://min-api.cryptocompare.com/data/price?fsym=WAXP&tsyms=BTC,USD";
	
}

const owner = process.env.ORACLE;
const oracleContract = process.env.CONTRACT;
const defaultPrivateKey = process.env.EOS_KEY;
const permission = process.env.ORACLE_PERMISSION;
const httpEndpoint =  process.env.EOS_PROTOCOL + "://" +  process.env.EOS_HOST + ":" + process.env.EOS_PORT;

console.log(owner,oracleContract,permission)
const rpc = new JsonRpc(httpEndpoint, { fetch });
const signatureProvider = new JsSignatureProvider([defaultPrivateKey]);
const api = new Api({ rpc, signatureProvider, textDecoder: new TextDecoder(), textEncoder: new TextEncoder() });


const eosmain = async (quotes2) => {
    try {
    const result = await api.transact({
        actions: [{
            account: oracleContract,
            name: 'write',
            authorization: [{
                actor: owner,
                permission: permission,
            }],
            data: {
                owner: owner,
                quotes: quotes2
            },
        }]
        }, {
            blocksBehind: 3,
            expireSeconds: 30,
        });
        console.dir(result);
    } catch (e) {
        console.log('\nCaught exception: ' + e);
        if (e instanceof RpcError)
          console.log(JSON.stringify(e.json, null, 2));
      }
  }

function writequotes(){
	axios
	.get(priceUrl)
	.then(response => {
		//Assign repsonse to quotes
		const quotes2 = [{"value": Math.round((response.data.BTC)* 100000000), pair: btcpair }, {"value": Math.round((response.data.USD)* 10000), pair: usdpair }]
		console.log(quotes2)
        //Call eos.contracts method 
        eosmain(quotes2)
	})
}

writequotes();

File: scripts/docker-updater.py
import json
import requests
import eospyo
import oracleconf as cfg
from pydantic import HttpUrl

EOS_PROTOCOL = cfg.main["EOS_PROTOCOL"]
EOS_HOST = cfg.main["EOS_HOST"]
EOS_PORT = cfg.main["EOS_PORT"]
EOS_KEY = cfg.main["EOS_KEY"]
EOS_CHAIN = cfg.main["EOS_CHAIN"]
ORACLE = cfg.main["ORACLE"]
CONTRACT = cfg.main["CONTRACT"]
ORACLE_PERMISSION = cfg.main["ORACLE_PERMISSION"]
FREQ = cfg.main["FREQ"]
CHAIN = cfg.main["CHAIN"]
APIKEY = cfg.main["APIKEY"]


class MyNetwork(eospyo.Net):
    host: HttpUrl = EOS_HOST 


#data = '[ 

auth = eospyo.Authorization(
    actor=ORACLE,
    permission=ORACLE_PERMISSION,
)


def getPrice():
    response = requests.get("https://min-api.cryptocompare.com/data/pricemulti?fsyms=USD,WAXP&tsyms=USDT,USDC,BTC,USD,ETH,EOS&api_key="+APIKEY)
    json_response = response.json()
    BTC = round(json_response['WAXP']['BTC']*100000000)
    USD = round(json_response['WAXP']['USD']*10000)
    EOS = round(json_response['WAXP']['EOS']*1000000)
    ETH = round(json_response['WAXP']['ETH']*100000000)
    USDUSDT = round(json_response['USD']['USDT']* 10000)
    USDUSDC = round(json_response['USD']['USDC']* 10000)
    data = [
        {'value': BTC, 'pair': 'waxpbtc' }, 
        {'value': USD, 'pair': 'waxpusd' },
        {'value': EOS, 'pair': 'waxpeos' },
        {'value': ETH, 'pair': 'waxpeth' }, 
        {'value': USDUSDT, 'pair': 'usdtusd' },
        {'value': USDUSDC, 'pair': 'usdcusd' }
    ]
    print(data)
    return data


pricedata = getPrice()

action = eospyo.Action(
    account=CONTRACT,  # this is the contract account
    name="write",  # this is the action name
    data={
            'owner': ORACLE,
            'quotes': pricedata
            },
    authorization=[auth],
)

raw_transaction = eospyo.Transaction(actions=[action])


print("Link transaction to the network")
#net = eospyo.WaxMainnet()
net = MyNetwork()
# notice that eospyo returns a new object instead of change in place
linked_transaction = raw_transaction.link(net=net)


print("Sign transaction")
key = EOS_KEY
signed_transaction = linked_transaction.sign(key=key)


print("Send")
resp = signed_transaction.send()

print("Printing the response")
resp_fmt = json.dumps(resp, indent=4)
print(f"Response:\n{resp_fmt}")
